packaging/_parser.py

```diff
@@ -0,0 +1,155 @@
+from typing import Any, List, Tuple
+
+from ._tokenizer import Tokenizer
+
+
+def parse_named_requirement(requirement: str) -> Tuple[Any, str, Any, str, str]:
```

Comment:
Shouldn't this actually be:

```suggestion
def parse_named_requirement(requirement: str) -> Tuple[str, str, str, str, str]:
```

? Similarily in all the other places where the type is set to Any.

Reply:
In this case, it should be `Tuple[str, str, List[str], str, str]`. I will take a look at the rest of them.